site/app/models/OfficeHoursQueueModel.php

```diff
@@ -130,6 +130,11 @@ public function getCurrentQueue() {
         return $this->current_queue;
     }
 
+    public function getDisplayNameFromUserId(string $user_id): string {
+        $user = $this->core->getQueries()->getUserById($user_id);
```

Comment:
Ah looks like this is causing the tests to detect excessive queries to fail, the model should have the current user cached so you can directly access it without have to query it each time.

see here for an example:
https://github.com/Submitty/Submitty/blob/main/site/app/models/OfficeHoursQueueModel.php#L102

so your function changes would probably be something like:

```suggestion
    public function getDisplayNameFromUserId(): string {
        $user = $this->core->getUser();
```

Reply:
Yeah, there are excessive queries happening. 

But I need the details of the user added to the queue using their `user id` and **not** the current user